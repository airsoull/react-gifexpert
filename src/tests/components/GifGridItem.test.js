import React from 'react';

const { shallow } = require('enzyme');
const { default: GifGridItem } = require('../../components/GifGridItem');

describe('Test for <GifGidItem/> component', () => {
  const title = 'foo-bar';
  const url = 'http://www.example.com/img.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should show component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a paragraph with the text', () => {
    expect(wrapper.find('p').text().trim()).toBe(title);
  });

  test('should have an img with url and title', () => {
    const imgProps = wrapper.find('img').get(0).props;
    expect(imgProps.src).toBe(url);
    expect(imgProps.alt).toBe(title);
  });

  test('should have a div with "card" class', () => {
    expect(wrapper.first('div').props().className).toBe('card');
  });
});
