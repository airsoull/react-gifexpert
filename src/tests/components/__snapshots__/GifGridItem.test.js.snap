// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <GifGidItem/> component should show component correctly 1`] = `
<div
  className="card"
>
  <div
    className="image-card"
  >
    <img
      alt="foo-bar"
      src="http://www.example.com/img.jpg"
    />
  </div>
  <p>
    foo-bar
  </p>
</div>
`;
